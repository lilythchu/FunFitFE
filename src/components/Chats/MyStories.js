import React, {useState, useEffect} from 'react';
import * as ImagePicker from 'expo-image-picker';
import {View, Text, Alert} from 'react-native';
import {LinearGradient} from 'expo-linear-gradient';
import {Avatar, ThemeProvider, Dialog} from '@rneui/themed';
import {avaGender} from '../../../utils/methods';
import client, {uploadStoryURL} from '../../../api/client';

const MyStories = ({token, navigation, userInfo}) => {
  const [uploadSuccess, setUploadSuccess] = useState(false);
  const [visibleDia, setVisibleDia] = useState(false);
  const [stories, setStories] = useState([]);
  
  const uploadImage = (uri) => {
    setVisibleDia(true);
    var myHeaders = new Headers();
    myHeaders.append('Authorization', `Bearer ${token}`);
    var formdata = new FormData();
    formdata.append('file', {
      uri: uri,
      name: 'file',
      type: 'image/png',
    });

    var requestOptions = {
      method: 'POST',
      headers: myHeaders,
      body: formdata,
    };

    fetch(uploadStoryURL, requestOptions)
      .then(response => response.text())
      .then(result => {
        setVisibleDia(false);
      })
      .catch(error => console.log(error));
  };

  const newStories = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [3, 4],
      quality: 1,
    });

    if (!result.cancelled) {
      uploadImage(result.uri);
    }
  };
  
  const getStoriesInfo = () => {
    client
      .get('/story/getStoriesInfo', {
        headers: {Authorization: `Bearer ${token}`},
        params: {id: userInfo._id},
      })
      .then(response => setStories(response.data))
      .catch(err => console.log(err));
  }

  const viewMyStories = () => {
    if (stories[0] === undefined) {
      Alert.alert("You don't have any stories");
    } else {
      navigation.navigate('MyStory', {stories})
    }
  }

  useEffect(() => getStoriesInfo(), [stories]);

  return (
    <LinearGradient
      colors={['#50b1f2', '#c7c432', '#32c790']}
      style={{padding: 3, borderRadius: 70}}>
      <Avatar
        size={64}
        rounded
        source={avaGender(userInfo.sex)}
        onPress={viewMyStories}
        containerStyle={{backgroundColor: 'grey'}}>
        <Avatar.Accessory
          name="plus"
          type="feather"
          size={23}
          onPress={newStories}
        />
      </Avatar>

      <ThemeProvider>
        <Dialog
          isVisible={visibleDia}
          overlayStyle={{borderRadius: 15}}>
          <Dialog.Title title="Uploading File..." />
        </Dialog>
      </ThemeProvider>
    </LinearGradient>
  )
};

export default MyStories;