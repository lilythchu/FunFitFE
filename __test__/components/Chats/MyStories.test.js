import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import MyStories from '../../../src/components/Chats/MyStories';

const mockUserInfo = {
  name: 'user1',
  sex: 'Female',
};

const mockNavigation = {
  navigate: jest.fn()
}

const setState = jest.fn();
const useStateSpy = jest.spyOn(React, 'useState');
useStateSpy.mockImplementation(initialState => [initialState,setState]);

describe('My Stories', () => {
  beforeEach(() => {
    render(<MyStories userInfo={mockUserInfo} navigation={mockNavigation} />)
  });

  test('renders correctly', () => {
    expect(screen).toMatchSnapshot();
  });

  // test('navigate to My Story screen if having stories', () => {
  //   fireEvent.press(screen.getByTestId('myStories'));
  //   expect(mockNavigation.navigate).toHaveBeenCalled();
  // })

});
