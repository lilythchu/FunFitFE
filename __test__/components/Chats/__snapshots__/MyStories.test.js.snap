// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Stories renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4283478514,
        4291281970,
        4281517968,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "borderRadius": 70,
      "padding": 3,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "grey",
        "borderRadius": 32,
        "height": 64,
        "opacity": 1,
        "width": 64,
      }
    }
    testID="myStories"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 32,
          "flex": 1,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        borderRadius={32}
        onLoad={[Function]}
        source={1}
        style={
          Object {
            "bottom": 0,
            "flex": 1,
            "height": undefined,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": undefined,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        collapsable={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "height": undefined,
              "justifyContent": "center",
              "width": undefined,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#aaa",
            "bottom": 0,
            "justifyContent": "center",
            "position": "absolute",
            "right": 0,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 2,
          },
          Object {
            "borderRadius": 11.5,
            "height": 23,
            "width": 23,
          },
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            testID="addStory"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    replaceTheme={[Function]}
    testID="Internal__Overlay"
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
    transparent={true}
    updateTheme={[Function]}
    visible={false}
  />
</ViewManagerAdapter_ExpoLinearGradient>
`;
